src/main/java/org/wmn4j/mir/MonophonicPattern.java

```diff
@@ -104,7 +104,21 @@ public boolean equalsInTransposedPitch(Pattern other) {
 
 	@Override
 	public boolean equalsInDurations(Pattern other) {
-		// TODO Auto-generated method stub
+		if (other.isMonophonic()) {
+			final List<Durational> contentsOfThis = getContents();
+			final List<Durational> contentsOfOther = other.getContents();
+
+			if (contentsOfThis.size() == contentsOfOther.size()) {
```

Comment:
Does this lead to a wrong answer if the contents are unequal in size? This looks like it's going to return true, but shouldn't it return false?

Reply:
The return true statement should be inside the if-block.

Reply:
In fact there are other issues with this bit of code as well, I should improve the tests a bit and fix this :)

Reply:
Cool :+1:

Reply:
The return statement is now in the right place so that patterns of different length cannot be considered equal in durations. Also the method should compare only durations of notes to durations of notes and durations of rests to those of other rests. For example, if the first element of a pattern A is a note with quarter duration, then pattern B needs to also start with a note of quarter duration to be equal to A in durations. This has been fixed and test cases for different length patterns and patterns where notes and rests are not in same positions have been added.

Reply:
Looks good, but now another case came to my mind :smile: What if two patterns are identical, but the other has an eighth note rest where the other has two sixteenth note rests? I think those should be seen as equal. Maybe there should be some kind of trim method in Pattern that adds adjacent rests together (and maybe cuts off rests from the end and beginning)? That way you don't need to change this code very much, just trim the patterns in the beginning of the method :stuck_out_tongue:

Reply:
The reason why I went with a strict equality of durations in this is that there are also cases where the rhythms of patterns can be perceptually the same, even though the durations of notes are not the same, e.g., in the case where one pattern has a quarter note and the other has an eight note followed by an eightnote rest. This kind of comparison would work in the general case by comparing just the onset times of the notes. There was a method in the `Pattern` interface for this purpose, but in order to keep the scope of this issue down, it was removed and left to be implemented in the future.

Reply:
Ok, that makes sense. In that case I think this pr is ready to be merged :stuck_out_tongue: